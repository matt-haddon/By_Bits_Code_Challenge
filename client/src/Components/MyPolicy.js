import React, { useState, useEffect } from 'react';

import { Button } from '@material-ui/core';
import { useHistory } from 'react-router-dom';
import { IoIosCar } from 'react-icons/io';
import { RiCompassDiscoverFill } from 'react-icons/ri';
import { BiBarcode } from 'react-icons/bi';
import { FaAddressCard } from 'react-icons/fa';
import Car from '../Assets/car.gif';

const MyPolicy = ({ setData, data, setError }) => {
    const [loading, setLoading] = useState(true);
    const history = useHistory();

    useEffect(() => {
        setTimeout(() => {
            setLoading(false);
        }, 2000);
    }, []);

    const logOut = () => {
        setData('');
        setError(false);
        history.push('/');
    };

    if (loading) {
        return (
            <div className="loading">
                <img src={Car} alt="loading" />
                <p data-testid="loading">Loading...</p>
            </div>
        );
    } else {
        return (
            <div className="policy">
                <h1>My Policy</h1>
                <section className="data">
                    <div>
                        <BiBarcode />
                        <section>
                            <h6>Policy reference</h6>
                            <p data-testid="policy-ref">
                                {data.policy.policy_ref}
                            </p>
                        </section>
                    </div>
                    <div>
                        {' '}
                        <RiCompassDiscoverFill />
                        <section>
                            <h6>Cover type</h6>
                            <p data-testid="policy-cover">
                                {data.policy.cover}
                            </p>
                        </section>
                    </div>
                    <div>
                        <IoIosCar />
                        <section>
                            <h6>Car</h6>
                            <p data-testid="car">{`${
                                data.vehicle.make.charAt(0).toUpperCase() +
                                data.vehicle.make.slice(1)
                            } ${data.vehicle.model} ${
                                data.vehicle.colour.charAt(0).toUpperCase() +
                                data.vehicle.colour.slice(1)
                            } - ${data.vehicle.reg}`}</p>
                        </section>
                    </div>
                    <div>
                        <FaAddressCard />
                        <section>
                            <h6>Address</h6>
                            <p data-testid="address">{`${data.policy.address.line_1}, ${data.policy.address.line_2}, ${data.policy.address.postcode}`}</p>
                        </section>
                    </div>
                    <Button
                        data-testid="log-out"
                        onClick={logOut}
                        color="primary"
                    >
                        Log Out
                    </Button>
                </section>
            </div>
        );
    }
};

export default MyPolicy;
