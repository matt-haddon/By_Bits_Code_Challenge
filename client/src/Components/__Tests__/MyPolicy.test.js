import MyPolicy from '../MyPolicy';
import { render, cleanup, act } from '@testing-library/react';

afterEach(cleanup);

test('policy details and log out button should render', async () => {
    const data = {
        policy: {
            cover: 'comprehensive',
            policy_ref: 'fakeRef',
            address: {
                line_1: 'line 1',
                line_2: 'line 2',
                postcode: 'postcode',
            },
        },
        vehicle: {
            make: 'fakeMake',
            model: 'fakeModel',
            reg: 'fakeReg',
            colour: 'black',
        },
    };

    const container = document.createElement('div');

    const { getByTestId } = render(<MyPolicy data={data} />, container);
    const promise = new Promise((r) => setTimeout(r, 2000));
    await act(() => promise);
    const policyRef = getByTestId('policy-ref');
    const policyCover = getByTestId('policy-cover');
    const car = getByTestId('car');
    const signOut = getByTestId('log-out');

    expect(policyRef).toBeInTheDocument();
    expect(policyCover).toBeInTheDocument();
    expect(car).toBeInTheDocument();
    expect(signOut).toBeInTheDocument();
});

test('snapshot', () => {
    const { container } = render(<MyPolicy />);

    expect(container.firstChild).toMatchSnapshot();
});
