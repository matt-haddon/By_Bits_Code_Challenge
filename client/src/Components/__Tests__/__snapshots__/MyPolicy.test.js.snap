// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="loading"
>
  <img
    alt="loading"
    src="car.gif"
  />
  <p
    data-testid="loading"
  >
    Loading...
  </p>
</div>
`;
